### PR TITLE
Sailor alpha version

### PR DESCRIPTION
**Pull request by ellenaua**

### CODE REVIEW

File: Procfile [hunk 1]
```diff
@@ -0,0 +1 @@
+web: node run.js
```
Comment (drobiazko) [inline Procfile]: Let's change ``web`` to something else as it is a special process type at heroku.

File: lib/settings.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+    result.MONGO_URI = envVars.MONGO_URI || throwError('MONGO_URI is missing');
+    result.AMQP_URI = envVars.AMQP_URI || throwError('AMQP_URI is missing');
+    result.COMPONENT_PATH = envVars.COMPONENT_PATH || '';
+
+    result.REBOUND_QUEUE_TTL = 10 * 60 * 1000; // 10 min
+    result.REBOUND_INITIAL_EXPIRATION = envVars.REBOUND_INITIAL_EXPIRATION || 15000;
+    result.REBOUND_LIMIT = envVars.REBOUND_LIMIT || 2;
+
+    result.INCOMING_MESSAGES_QUEUE = {
```
Comment (drobiazko) [inline lib/settings.js]: Sailor must not be responsible to define any queue settings.

File: lib/amqp.js [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+        debug('Open publish channel');
+        return self.amqp.createChannel().then(function(channel) {
+            debug('Opened publish channel');
+            self.publishChannel = channel;
+        });
+    }
+
+    function assertAllQueues() {
+        return Q.all([
+            assertQueue(self.subscribeChannel, settings.INCOMING_MESSAGES_QUEUE),
```
Comment (drobiazko) [inline lib/amqp.js]: I think the assertion must be done outside of sailor.

File: lib/executor.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+    scope.on('end', onEnd);
+    function onEnd(){
+        clearTimeout(taskexec.executionTimeout);
+        deferred.resolve();
+        taskexec.emit('end');
+    }
+
+    try {
```
Comment (pnedelko) [inline lib/executor.js]: I guess we need to use asynchronous try-catch here. I saw the `trycatch` module has been used in some places in our current runtime.

File: Procfile (hunk 1 above)
Comment (drobiazko) [inline Procfile]: See here: https://devcenter.heroku.com/articles/procfile

File: lib/amqp.js (hunk 3 above)
Comment (drobiazko) [inline lib/amqp.js]: What I mean here is: Does it make sense sailor is not responsible for the queue configuration and creation? Shouldn't it happen somewhere else?

Comment (ellenaua) [inline lib/amqp.js]: It's OK, so sailor will only connect.. 
But now we have no that "entity" yet. 
Maybe I'll just make a script which creates queues (to use & test on localhost)?

File: lib/settings.js (hunk 2 above)
Comment (ellenaua) [inline lib/settings.js]: Got it

File: Procfile (hunk 1 above)
Comment (ellenaua) [inline Procfile]: "sail"?

File: README.md [hunk 5]
```diff
@@ -1,2 +1,27 @@
...
+AMQP_URI=amqp://guest:guest@localhost:5672
+TASK_ID=task123
+STEP_ID=step_2
+
+Default values of .env vars:
+
+COMPONENT_PATH=../
+INCOMING_MESSAGES_QUEUE={TASK_ID}.{STEP_ID}.incoming
+OUTGOING_MESSAGES_QUEUE={TASK_ID}.{STEP_ID}.outgoing
+ERRORS_QUEUE={TASK_ID}.{STEP_ID}.errors
```
Comment (skarbovskiy) [inline README.md]: I thought that there will be one queue for all errors.. just for simplicity. Why per step?

File: lib/executor.js (hunk 4 above)
Comment (skarbovskiy) [inline lib/executor.js]: +1

File: lib/amqp.js (hunk 3 above)
Comment (drobiazko) [inline lib/amqp.js]: We can create them on task creation.

File: README.md (hunk 5 above)
Comment (drobiazko) [inline README.md]: Do you want one error queue per task?

**Pull request closed without merging**